### PR TITLE
Feature/update cart flow

### PR DESCRIPTION
**Pull request by VladislavMinchuk**
Update cart.slice reducers:
- add functions for useSelect hook
- update initial state
- update extraReducers

Update ProductCard cmp:
- add new prop = show 'order' button for the product exist in user cart obj.

### CODE REVIEW

File: src/pages/CartPage.tsx [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 
 const CartPage: React.FC = () => {
-  const dispatch = useAppDispatch();
-  const userCart = useTypedSelector((state) => state.cartSlice.userCart);
-  const isLoading = useTypedSelector((state) => state.global.isLoading);
+  const dispatch      = useAppDispatch();
+  const totalPayment  = useTypedSelector(({ cartSlice }) => getTotalPayment(cartSlice));
+  const productList   = useTypedSelector(({ cartSlice }) => getProductList(cartSlice));
+  const userCartError = useTypedSelector(({ cartSlice }) => cartSlice.error);
+  const isLoading     = useTypedSelector(({ global }) => global.isLoading);
```
Comment (G1Lroy) [inline src/pages/CartPage.tsx:16]: Why not reduce the number of selector calls by getting all the necessary data from one line?)
` const {error, userCart: { products, payment }} = useTypedSelector((state) => state.cartSlice);`
besides, there will be no need to create getter functions in the slice.

p.s
_good alignment!_

File: src/pages/CartPage.tsx
Comment (G1Lroy) [inline src/pages/CartPage.tsx:1]: In general, I understand the desire to get rid of chains of access to object fields. (userCart?.payment.total). But not completely 🤔🤔

File: src/pages/CartPage.tsx [hunk 2]
```diff
@@ -22,7 +24,17 @@ const CartPage: React.FC = () => {
 
   useEffect(() => {
     dispatch(updateTotalPayment());
-  }, [userCart]);
+  }, [productList]);
+
+  useEffect(() => { // TODO: move to <App> ?
+    dispatch(getUserCart(1))    
+  }, []);
+
```
Comment (G1Lroy) [inline src/pages/CartPage.tsx:32]: If I understand correctly, before this change we did not call `getUserCart` ?

File: src/pages/CartPage.tsx (hunk 1 above)
Comment (VladislavMinchuk) [reply src/pages/CartPage.tsx:16]: yeah. totally agree.
I have read it in the documentation about slices and decided to use. I propose to leave it here as example how we can use selection from slice. You can select store item as you want :)

File: src/pages/CartPage.tsx (hunk 2 above)
Comment (VladislavMinchuk) [reply src/pages/CartPage.tsx:32]: yes. we didn't have this dispatch before.
also we should decide the place in app where we can call 'general dispatch' for getting users data, users cart etc.

File: src/pages/CartPage.tsx
Comment (VladislavMinchuk) [reply src/pages/CartPage.tsx:1]: Do you mean destructure of function arguments?
it doesn't have any profit. It's just for getting exactly field what I need to

**Review by G1Lroy — APPROVED** [review]

**Pull request merged**